### PR TITLE
Implemented some of the button functionalities to the GUI + added solid motor pressure csv

### LINKED ISSUE #3: Implemente a função Reset
Quando se clicar na função reset, deve-se limpar o painel e limpar os dados armazenados.

### LINKED ISSUE #6: Instrumentação do sólido
Para a instrumentação do sólido implementei os seguintes botões:

- Reset, reseta o estado do sistema
- Ignition, manda sinal de ignição para o microcontrolador
- Save, Salva os dados

Você adicionaria mais algum? Ve alguma funcionalidade faltando?

### LINKED ISSUE #14: Implementar plot da pressão do Sólido
- Adicionar um Mock .csv da curva de pressão do sólido, proveniente de algum dos outros códigos da Prop
- Alterar o código conforme o necessário de modo a plotar esses dados corretamente, simultaneamente aos dados de Empuxo

A intenção é pensar em possíveis modificações a serem feitas a partir do momento em que o código recebe informações de várias fontes (no momento a conexão recebe somente inputs de força do static.csv)

### PR DESCRIPTION
**Pull request by MrGribel**
- Added a pressure data .csv derived from the current thrust list, derived from one of the team's codes

Also tried to implement functions to some of the GUI preexisting buttons. More specifically:
- Added a 'START/STOP button' individually for each of the sensors, that works by turning on/off the update function of each.
- The 'Reset' button now erases all unsaved data and clears the plot windows
- Created a function that adds a prompt window to important buttons such as 'Reset' and 'Ignition' (making sure you don't press then by accident)

### CODE REVIEW

**Comment by MrGribel** [comment]
However, the code still currently only handles a single connection at a time (at the moment, it connects to a mock which handles the thrust data from the .csv).
This prevents us from testing situations in which we would receive different inputs at the same time (e.g. Pressure and Thrust information coming in, we would like to plot them both at the same time)
@PatrickSampaioUSP do you have any suggestions?

File: ButtonFunctions/__init__.py [hunk 1]
```diff
@@ -19,3 +23,35 @@ def save_curve(application) -> None:
...
+        application.data_initialization()
+        for key, item in application.plot_pannels.items():
+            item.clear()
+
+
+
+
+
+def start_transducer(application):
+    sensor = application.get_sensor_info(PRESSURE_TRANSDUCER_REF)
```
Comment (PatrickSampaioUSP) [inline ButtonFunctions/__init__.py]: Can we make this kind of switch generic? Because we are going to have a lot of functions similar to those

**Comment by PatrickSampaioUSP** [comment]
I will download this code into my machine to analyse it carefully, but sounds very good to me. Also, could you add docstrings to this new class?

File: ButtonFunctions/__init__.py (hunk 1 above)
Comment (MrGribel) [reply ButtonFunctions/__init__.py]: Great idea. I thought about it a while ago.
Will work on it.

**Comment by MrGribel** [comment]
> 
> 
> I will download this code into my machine to analyse it carefully, but sounds very good to me. Also, could you add docstrings to this new class?

I will look into it. The code definitely needs a bit more documentation, especially on the recent changes.

**Comment by PatrickSampaioUSP** [comment]
@MrGribel Your change proposal is very interesting, I think that our application is going to an interesting direction. Keep up with the good work!

I made some changes, firstly I don't think that Application should inherit from Data, because it's a class dedicated to handling dictionaries, plots and data related stuff, therefore i think a sensor data should be a instance of the Data object, but I don't imagine why Application would inherit Data. If you can explain me what you though we can discuss that point. Secondly I redefined the plot_pannels dictionary in the Application layer, cause I think that the class attribute you made was easily a instance attribute, if you think that there is a strong motive for it being a class attribute we can revise that point. Please revise my changes and tell me if you agree.

**Comment by PatrickSampaioUSP** [comment]
Also, could you add documentaiton to the "data_switch" method? I think it's really well done, but it lacks documentation.

**Comment by MrGribel** [comment]
> I don't imagine why Application would inherit Data.

It was a workaround so that the `plot_pannels` dictionary could be accessed by other functions that receive the `application` instance such as `Update` and the `ButtonFunctions`.

> I redefined the plot_pannels dictionary in the Application layer

Perfect! That was in fact my original idea, I later moved them to a class attribute in `Data` class following your suggestion that we should move the pannels. But I totally agree that the way it is now is way more simpler and intuitive. The pannels themselves are a `PlotItem` object from PyQt, so naturally they should be handled by the Application itself. This way we also avoid the need for inheritance between the classes.

I will review if everything is ok with the resolved conflicts and try documenting the missing parts of the code. Then, if everything is good we can merge this PR and start working on the remaining issues of the project.

**Pull request merged**